src/main/java/net/minecraftforge/fmp/util/IWorldLocation.java

```diff
@@ -0,0 +1,16 @@
+package net.minecraftforge.fmp.util;
+
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+/**
+ * Representation of the {@link World} and {@link BlockPos} of an object.
+ */
+public interface IWorldLocation
+{
+    
+    public World getWorldIn();
```

Comment:
We'd need something like @RainWarrior's tree obfuscator if we want something like this to work without custom names (or at least that's what he told me). Right now it has "in" at the end to prevent the names from matching the ones in classes like TileEntity and breaking obfuscation.